pallets/crowdloans/src/lib.rs

```diff
@@ -240,6 +252,37 @@ pub mod pallet {
             Ok(())
         }
 
+        /// Update an exisiting vault via a governance decision
+        #[pallet::weight(<T as Config>::WeightInfo::update_vault())]
+        #[transactional]
+        pub fn update_vault(
+            origin: OriginFor<T>,
+            crowdloan: ParaId,
+            cap_limit: Option<BalanceOf<T>>,
+            end_block: Option<BlockNumber>,
+            contribution_strategy: Option<ContributionStrategy>,
+        ) -> DispatchResult {
+            T::UpdateVaultOrigin::ensure_origin(origin)?;
+
+            let mut vault = Self::current_vault(crowdloan).ok_or(Error::<T>::VaultDoesNotExist)?;
+
+            if let Some(cap_limit) = cap_limit {
+                vault.cap_limit = cap_limit;
```

Comment:
can we simplify `cap_limit` to be `cap`? because cap and limit may have similar meanings